cmd/manager/os.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"os"
+	"os/user"
+	"path/filepath"
+	"time"
+
+	"github.com/docker/infrakit/discovery"
+	file_leader "github.com/docker/infrakit/leader/file"
+	file_store "github.com/docker/infrakit/store/file"
+	"github.com/spf13/cobra"
+)
+
+const (
+	// LeaderFileEnvVar is the environment variable that may be used to customize the plugin leader detection
+	LeaderFileEnvVar = "INFRAKIT_LEADER_FILE"
+
+	// StoreDirEnvVar is the directory where the configs are stored
+	StoreDirEnvVar = "INFRAKIT_STORE_DIR"
+)
+
+func getHome() string {
+	home := os.Getenv("HOME")
+	if usr, err := user.Current(); err == nil {
+		home = usr.HomeDir
+	}
+	return home
+}
+
+func defaultLeaderFile() string {
+	if leaderFile := os.Getenv(LeaderFileEnvVar); leaderFile != "" {
+		return leaderFile
+	}
+	return filepath.Join(getHome(), ".infrakit/leader")
+}
+
+func defaultStoreDir() string {
+	if storeDir := os.Getenv(StoreDirEnvVar); storeDir != "" {
+		return storeDir
+	}
+	return filepath.Join(getHome(), ".infrakit/configs")
+}
+
+func osEnvironment(backend *backend) *cobra.Command {
+
+	pollInterval := 5 * time.Second
+	filename := defaultLeaderFile()
+	storeDir := defaultStoreDir()
```

Comment:
I hear you, and I don't think the design of the cobra API makes it easy to structure well.

Another problem resulting from the initializing approach here is you have two places to check for the default - the variable _and_ flag definition to ensure that it was wired into the default parameter.  I've had to correct a few places already where the latter was missing, leading to confusion.